StateOfChargeManager.h

```diff
@@ -0,0 +1,38 @@
+#ifndef STATEOFCHARGEMANAGER
+#define STATEOFCHARGEMANAGER
+
+class StateOfChargeManager : public BatteryManagementSystem
+{
+    public:
+    bool checkOptimumLimit(float stateOfCharge)

```

Comment:
Name functions that return `bool` with 'is' in them. Like `isWithinLimit`.
Check that they read better in an if-statement: `if (isWithinLimit...`

Reply:
Changed to isWithinLimit.